### ISSUE TITLE
Modularize WebGL backend methods

### ISSUE THREAD

**Issue by tafsiri** [tensorflow/tfjs#4095]
### List of methods on the backend.ts interface. 

These are methods that need to be modularized and removed from the backend class.

- [x] batchMatMul (annxingyuan)
- [x] fusedBatchMatMul (annxingyuan)
- [x] slice
- [x] stridedSlice (annxingyuan)
- [x] unstack (yannick)
- [x] reverse (annxingyuan)
- [x] concat
- [x] neg (matthew)
- [x] add
- [x] addN (yannick)
- [x] subtract
- [x] multiply
- [x] realDivide
- [x] floorDiv (yannick)
- [x] sum
- [x] prod (yannick)
- [x] unsortedSegmentSum (yannick)
- [x] argMin (annxingyuan)
- [x] argMax (annxingyuan)
- [x] equal (yannick)
- [x] notEqual
- [x] less (yannick)
- [x] lessEqual (yannick)
- [x] greater (yannick)
- [x] greaterEqual (yannick)
- [x] logicalNot (matthew)
- [x] logicalAnd (yannick)
- [x] logicalOr (yannick)
- [ ] where (yannick)
- [x] select (yannick)
- [x] topk (yannick)
- [x] min (yannick)
- [x] minimum (yannick)
- [x] mod (yannick)
- [x] max
- [x] maximum (yannick)
- [x] all (yannick)
- [x] any (yannick)
- [x] squaredDifference
- [x] ceil (matthew)
- [x] floor (mattsoulanille)
- [x] round (matthew)
- [x] sign (mattsoulanille)
- [x] isNaN (matthew)
- [x] isInf (matthew)
- [x] isFinite (matthew)
- [x] pow
- [x] exp (yannick)
- [x] expm1 (matthew)
- [x] softmax (yannick)
- [x] log (matthew)
- [x] log1p (matthew)
- [x] sqrt (matthew)
- [x] rsqrt (matthew)
- [x] square
- [x] reciprocal (matthew)
- [x] relu (matthew)
- [x] relu6 (matthew)
- [x] prelu
- [x] elu (matthew)
- [x] eluDer (yannick)
- [x] selu (matthew)
- [x] int
- [x] clip (yannick)
- [x] abs (matthew)
- [x] complexAbs (yannick)
- [x] sigmoid
- [x] softplus
- [x] sin
- [x] cos
- [x] tan
- [x] asin
- [x] acos
- [x] atan
- [x] atan2
- [x] sinh
- [x] cosh
- [x] tanh
- [x] asinh
- [x] acosh
- [x] atanh
- [x] erf
- [x] step (matthew)
- [x] fusedConv2d
- [x] conv2d
- [x] conv2dDerInput (annxingyuan)
- [x] conv2dDerFilter (annxingyuan)
- [x] fusedDepthwiseConv2D (annxingyuan)
- [x] depthwiseConv2D (annxingyuan)
- [x] depthwiseConv2DDerInput (annxingyuan)
- [x] depthwiseConv2DDerFilter (annxingyuan)
- [x] conv3d (annxingyuan)
- [x] conv3dDerInput (annxingyuan)
- [x] conv3dDerFilter (annxingyuan)
- [x] maxPool
- [x] maxPoolBackprop
- [x] avgPool
- [x] avgPoolBackprop
- [x] avgPool3d (annxingyuan)
- [x] avgPool3dBackprop (annxingyuan)
- [x] maxPool3d (annxingyuan)
- [x] maxPool3dBackprop (annxingyuan)
- [x] reshape
- [x] cast
- [x] tile
- [x] pad (yannick)
- [x] transpose
- [x] gather (yannick)
- [x] gatherND (yannick)
- [x] scatterND (yannick)
- [x] batchToSpaceND (yannick)
- [x] spaceToBatchND (yannick)
- [x] resizeBilinear (yannick)
- [x] resizeBilinearBackprop (yannick)
- [x] resizeNearestNeighbor (yannick)
- [x] resizeNearestNeighborBackprop (yannick)
- [x] batchNormalization
- [x] localResponseNormalization4D (yannick)
- [x] LRNGrad (yannick)
- [x] multinomial (yannick)
- [x] oneHot (yannick)
- [x] cumsum (yannick)
- [x] nonMaxSuppression
- [x] fft
- [x] ifft
- [x] complex
- [x] real
- [x] imag
- [x] cropAndResize (yannick)
- [x] depthToSpace (yannick)
- [x] split (yannick)
- [x] sparseToDense (yannick)
- [x] diag (yannick)
- [x] fill (yannick)
- [x] onesLike (yannick)
- [x] zerosLike (yannick)
- [x] linspace (yannick)

**Comment by annxingyuan** [comment]
Closing as this has been completed!
